internal/controller/helpers_test.go

```diff
@@ -0,0 +1,159 @@
+package controller
+
+import (
+	"strings"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+func Test_findEnvVarIndex(t *testing.T) {
+	type args struct {
+		envVarName string
+		envVarList []corev1.EnvVar
+	}
+	tests := []struct {
+		name string
+		args args
+		want int
+	}{
+		{
+			name: "correctly finds the index of the env var",
+			args: args{
+				envVarName: "test",
+				envVarList: []corev1.EnvVar{
+					{
+						Name:  "test",
+						Value: "test",
+					},
+				},
+			},
+			want: 0,
+		},
+		{
+			name: "correctly finds the index of the env var",
```

Comment:
I suppose this test represent a scenario where the arg envVarName is not in envVarList and therefore function return -1 , right ? in that case it's still true to name it "correctly finds the index of the env var"  ? maybe test envVarName does not exist or something like that